hail/src/main/scala/is/hail/expr/ir/InferType.scala

```diff
@@ -31,6 +31,7 @@ object InferType {
         TNDArray(coerce[TArray](data.typ).elementType.setRequired(true), Nat(shape.typ.asInstanceOf[TTuple].size))
       case _: ArrayLen => TInt32()
       case _: StreamRange => TStream(TInt32())
+      case _: ArrayZeros => TArray(TInt32())
```

Comment:
This element type could be required, until InferPTypes is used. We'll removed the requiredeness property in the next week or so, but that would allow codegen for ArrayZeros to be correct for an extra week (namely the lack of need to store the missingness header0

Reply:
I'm inclined to just let it be missing until InferPTypes works correctly. My experience with NDArrays suggests random requiredness virtual types things will break.

Reply:
That is understandable given your experiences with requiredeness, which Tim and I will fix.